pkg/kubelet/kubelet.go

```diff
@@ -257,7 +257,8 @@ type DockerOptions struct {
 
 // makePodSourceConfig creates a config.PodConfig from the given
 // KubeletConfiguration or returns an error.
-func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, kubeDeps *Dependencies, nodeName types.NodeName) (*config.PodConfig, error) {
+// If a client is available it would also return a Reflector for apiserver pod sources.
+func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, kubeDeps *Dependencies, nodeName types.NodeName, nodeHasSynced func() bool) (*config.PodConfig, *cache.Reflector, error) {
```

Comment:
Why is this both passing in a `nodeHasSynced func() bool` and passing back an unstarted reflector?

Plumbing `nodeHasSynced` to `NewSourceApiserver` and doing the for loop on that inside a spawned goroutine there before calling r.Run seems like it would be cleaner (and easier to unit test).

Reply:
> Why is this both passing in a nodeHasSynced func() bool and passing back an unstarted reflector?

oversight on my end.

updated to loop in NewSourceApiserver until calling newSourceApiserverFromLW.
one downside of this is that one go routine in NewSourceApiserver would spawn another one in newSourceApiserverFromLW for the r.Run().

alternatively nodeHasSynced can plumb to newSourceApiserverFromLW as well , but this means that unit tests that call newSourceApiserverFromLW directly would need a stub of e.g. `nodeHasSynced := func() bool { return true }`.